app/server/lib/DocApi.ts

```diff
@@ -720,7 +735,9 @@ export class DocWorkerApi {
     // Add or update records given in records format
     this._app.put('/api/docs/:docId/tables/:tableId/records', canEdit, validate(RecordsPut),
       withDoc(async (activeDoc, req, res) => {
-        const ops = getTableOperations(req, activeDoc);
+        const metaTables = await getMetaTables(activeDoc, req);
+        const tableId = getRealTableId(metaTables, req.params.tableId);
+        const ops = getTableOperations(req, activeDoc, tableId);
```

Comment:
Please refactor `getTableOperations` to take care of this itself, since it already has the arguments for `getMetaTables`. Then these lines don't need to change.

I also think `getRealTableId` should accept either `metaTables` (for when they have multiple uses) or `activeDoc, req` to fetch them itself. And it should only fetch them after checking that `tableId` is a number, since it's not free and won't be needed most of the time.

Reply:
Or just put the function `getMetaTables` out of the `DocWorkerApi` and make it exportable

Reply:
I think just put getRealTableId in DocApi.